### PR TITLE
Use fully qualified name for all macro non-prelude types

### PR DESCRIPTION
**Pull request by bgurney-rh**
Resolves: #522

Supercedes: #523

### CODE REVIEW

**Comment by jbaublitz** [comment]
I'm just catching up on this now. `Debug` and `Hash` are actually both considered "special" when it comes to procedural macros (the `#[derive(...)]` attributes). Both of these do not need to be fully qualified by their absolute path as neither of these derive statements introduces a dependency on a `use` outside of the macro. Because both of these are registered with the compiler as specific code generation keywords and procedural macros built by the core Rust team seem use absolute paths pretty consistently, this no longer becomes a symbol as much as the name of the code generation method to call at compile time. Because these methods are built in and not custom derives (see [here](https://doc.rust-lang.org/book/ch19-06-macros.html#how-to-write-a-custom-derive-macro) for more information on custom derives), they do not need a path when specified in a derive. `use` is only needed outside of the `#[derive(...)]` statement in custom derives when importing a procedural macro from another crate and even here, because it is outside of the generated code, will be necessary as a `use` and does not need to be avoided.

File: src/id_macros.rs [hunk 1]
```diff
@@ -35,21 +35,21 @@ macro_rules! str_check {
 macro_rules! str_id {
     ($B:ident, $O:ident, $MAX:ident, $err_func:ident) => {
         /// The borrowed version of the DM identifier.
-        #[derive(Debug, PartialEq, Eq, Hash)]
+        #[derive(std::fmt::Debug, PartialEq, Eq, std::hash::Hash)]
```
Comment (jbaublitz) [inline src/id_macros.rs]: Please change these to `Debug` and `Hash` per my comment on the PR.

File: src/id_macros.rs [hunk 2]
```diff
@@ -35,21 +35,21 @@ macro_rules! str_check {
...
         pub struct $B {
             inner: str,
         }
 
         /// The owned version of the DM identifier.
-        #[derive(Debug, Clone, PartialEq, Eq, Hash)]
+        #[derive(std::fmt::Debug, Clone, PartialEq, Eq, std::hash::Hash)]
```
Comment (jbaublitz) [inline src/id_macros.rs]: Please change these to Debug and Hash per my comment on the PR.

File: src/id_macros.rs (hunk 1 above)
Comment (bgurney-rh) [reply src/id_macros.rs]: Resolved.

File: src/id_macros.rs (hunk 2 above)
Comment (bgurney-rh) [reply src/id_macros.rs]: Resolved.

**Pull request merged**